.github/workflows/ci.yml

```diff
@@ -110,3 +112,39 @@ jobs:
         timeout_minutes: 15
         max_attempts: 2
         command: ./tools/test_monitor/run-tests.sh
+  flow-legacy-client-compatibility-test:
+    name: Client Compatibility Tests With Observer
+    strategy:
+      fail-fast: false
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ env.GO_VERSION }}
+    - name: Checkout repo
+      uses: actions/checkout@v2
+    - name: Build relic
+      run: make crypto_setup_gopath
+    - name: Docker build
+      run: make install-tools
+    - name: Install Flow Client
+      run: sh -ci "$(curl -fsSL https://storage.googleapis.com/flow-cli/install.sh)"
+    - name: Set up Localnet
+      run: bash -c 'cd integration/localnet/ && make -e OBSERVER=1 init && make start'
+    - name: Ensure Observer is started
+      run: docker ps -q -f name=localnet_observer_1_1
+    - name: Get Client Version ensuring the client is provisioned
+      run: ~/.local/bin/flow -f integration/localnet/flow-localnet.json -n observer version
+    - name: Wait for a default waiting period until a clean state
+      # This will not cause flakiness.
+      # The waiting time is a reasonable time to expect an observer to be responsive
+      run: sleep 10
+    - name: Get Status ensuring the observer endpoint is online
+      # observer reference will be replaced with access after gRPC proxying
+      run: ~/.local/bin/flow -f integration/localnet/flow-localnet.json -n access status -o json | grep '"status":"ONLINE"'
+    - name: Wait for finalized blocks and check them with Access
+      # observer reference will be replaced with access after gRPC proxying
```

Comment:
I'm not sure I follow this comment. Do you mean that this step will be updated to run against the observer instead of the access node once it has a working gRPC server?

Reply:
I had issues running clients against the observer in master simulated by localnet-access image.
I did not spend much time to investigate this intermittent issue, as the observer code added later worked.

I can fix the yaml now. It meant to stay on access to ensure the test is stable until we gradually add changes.
It is not needed anymore, and I can update it to observers, because we have this big pull request now.

Reply:
Fix: https://github.com/GetElastech/flow-go/commit/8380f81b0f3b56cafa4acedb3ffefde46af56ca4